fanduelOdds.php

```diff
@@ -35,13 +94,218 @@ private function getLiveMarketIds() {
+                    if ($runnerDetail->selectionId == self::RAPTORS_ID) {
+                        // grab the eventId of the Raptors game
+                        $raptorsEventId = $eventId;
+                    }
+                }
+            }
+            if (!isset($groupedMarketsByGame->$raptorsEventId)) {
+                return [];
+            }
+            else {
+                return $groupedMarketsByGame->$raptorsEventId;
+            }           
+        }
+    }
+
+    private function processMarketPrices($marketPrices) {
+        /**
+         * Helper function that groups market prices by Raptors, opponent, and Over/Under
+         *
+         * @return array Array of market prices with groupings listed above. [Raptors, Opponent, Over/Under]
+         */
+
+        $raptorsPrices = new stdClass();
+        $opponentPrices = new stdClass();
+        $overUnderPrices = new stdClass();
+
+        foreach ($marketPrices as $market) {            
+
+            if ($market->marketName == "Moneyline") {
+                foreach($market->runnerDetails as $runnerDetail) {
+                    if ($runnerDetail->selectionId == self::RAPTORS_ID) {
+                        $raptorsPrices->moneyLine = [$this->convertToAmericanOdds($runnerDetail)[0]];
+                    }
+                    else {
+                        $opponentPrices->moneyLine = [$this->convertToAmericanOdds($runnerDetail)[0]];
+                    }
+                }
+            }
+            elseif ($market->marketName == "Spread Betting") {
+                foreach($market->runnerDetails as $runnerDetail) {
+                    if ($runnerDetail->selectionId == self::RAPTORS_ID) {
+                        $raptorsPrices->spread = $this->convertToAmericanOdds($runnerDetail);
+                    }
+                    else {
+                        $opponentPrices->spread = $this->convertToAmericanOdds($runnerDetail);
+                    }
+                }
+            }
+
+            elseif ($market->marketName == "Total Points") {
+                foreach($market->runnerDetails as $runnerDetail) {
+                    if ($runnerDetail->selectionId == self::OVER_ID) {
+                        $overUnderPrices->over = $this->convertToAmericanOdds($runnerDetail, true);
+                    }
+                    else {
+                        $overUnderPrices->under = $this->convertToAmericanOdds($runnerDetail, true);
+                    }
+                }
+            }            
+        }
+        return [$raptorsPrices, $opponentPrices, $overUnderPrices];
+    }
+
+    private function getOpposingTeam($marketPrices) {
+        /**
+         * Helper function that grabs the opponent team name
+         *
+         * @return string Opponent team name
+         */
+
+        foreach ($marketPrices as $market) {      
+            if ($market->marketName != "Total Points") {   
```

Comment:
Is there a market id you can compare this with instead of the phrase "Total Points" as that seems a descriptive text more than a more permanent indicator?

Reply:
Will investigate

Reply:
There is a marketId, but that is for the specific game + marketType (Over/Under, Spread, Moneyline) combination from what I can tell

I can change it to the enum string however that was provided in the documentation